bpf/bpf_sock.c

```diff
@@ -201,9 +201,14 @@ static inline void sock4_handle_node_port(struct bpf_sock_addr *ctx,
 		return;
 
 	info = ipcache_lookup4(&IPCACHE_MAP, daddr, V4_CACHE_KEY_LEN);
-	if (info != NULL && (info->sec_label == HOST_ID ||
-			     info->sec_label == REMOTE_NODE_ID))
+	if (info != NULL && (info->sec_label == HOST_ID
+#ifdef ENABLE_NODEPORT_XLR
+			     || info->sec_label == REMOTE_NODE_ID)) {
+#else
+	)) {
+#endif
```

Comment:
Lets make this a proper helper function e.g. `sock_can_xlate_direct(info->sec_label)` and move the ifdef in there. This should be applied to NodePort and ExternalIP.

Reply:
Commit desc says: `To simplify the testing, introduce --enable-node-port-xlr flag (defaults
to "true") which optionally enables the optimization.` -> `optionally disables`